simulators/nav/src/components/perception/obstacle_detector.ts

```diff
@@ -238,11 +238,20 @@ export default class ObstacleDetector {
       angle = intervalHeap.maxOccupied + (minIntervalSize / 2);
     }
 
-    return {
-      detected: false,
-      distance: this.obsDist,
-      bearing: angle
-    };
+    if (this.obsDist >= 0) {
+      return {
+        detected: true, /* deprecated so always false */
+        distance: this.obsDist, /* Will be -1 if okay to go straight ahead (i.e. bearing = 0) */
+        bearing: angle
+      };
+    }
+    else {
+      return {
+        detected: false, /* deprecated so always false */
+        distance: this.obsDist, /* Will be -1 if okay to go straight ahead (i.e. bearing = 0) */
+        bearing: angle
+      };
+    }
```

Comment:
1. Can you add a comment signifying that we do not want to add this to our repository for real? In other words this should not be merged into neither `main` nor any `*-main` branches.
2. Why do we want to do this? Isn't the right answer to correct the programs misusing this field rather than hack the program correctly using it?
3. what about:
```
return {
    detected: this.obsDist >= 0,
    distance: this.obsDist,
    bearing: angle
};
```